pkg/server/server.go

```diff
@@ -607,9 +610,22 @@ func (s *Server) generatePolicyRules(pod *v1.Pod, podInfo *controllers.PodInfo)
 	}
 
 	if err := iptableBuffer.SyncRules(s.ip4Tables); err != nil {
-		klog.Errorf("sync rules failed: %v", err)
+		klog.Errorf("sync rules failed for pod [%s]: %v", namespacedName(pod), err)
 		return err
 	}
 
 	return nil
 }
+
+func namespacedName(obj runtime.Object) string {
```

Comment:
From logging perspective, I understand that `namespacedName()` for ease of debug message, however I suppose we should not use it here because golang `reflect` package affects the performance of the program as of following document. 

https://www.sobyte.net/post/2021-07/go-reflect-performance/

So could you come up with the function without `reflect` package, or find alternative approach for that?

Reply:
I agree using reflection is not a good choice from a performance point of view.
[Here](https://github.com/zeeke/multi-networkpolicy-iptables/blob/logging-benchmark/pkg/server/server_test.go) I created a comparative benchmark with the following result:

```
$ go test -bench=namespacedName ./pkg/server -benchmem -benchtime=20s                                                             
goos: linux
goarch: amd64
pkg: github.com/k8snetworkplumbingwg/multi-networkpolicy-iptables/pkg/server
cpu: Intel(R) Core(TM) i7-10610U CPU @ 1.80GHz
Benchmark_namespacedName_reflect-8                      152687607              157.7 ns/op            24 B/op          1 allocs/op
Benchmark_namespacedName_accessor_noreflect-8           172644097              138.6 ns/op            24 B/op          1 allocs/op
Benchmark_namespacedName_switch-8                       258165489               93.26 ns/op           24 B/op          1 allocs/op
Benchmark_namespacedName_direct-8                       519012726               45.72 ns/op            0 B/op          0 allocs/op
PASS
ok      github.com/k8snetworkplumbingwg/multi-networkpolicy-iptables/pkg/server 139.850s
```

Using a different function for each type (`Benchmark_namespacedName_direct`) makes the function three times faster and with no memory allocations. Honestly, I expected to discover a bigger gap between these implementations, but it may worth using it. 

`namespacedNameSwitch(...)` is based on a type switch and slightly increases the performance, I wouldn't go for this solution.

`namespacedNameAccessorNoReflect(...)`  [cannot be actually used](https://glucn.medium.com/golang-an-interface-holding-a-nil-value-is-not-nil-bb151f472cc7).

Any preference?

Reply:
Done!